add openid login to backstage bubble

jdlrobson opened the pull request:
TODO: we probably want to hide this for the average user. This at
least makes it easy to access this login form.

jdlrobson committed:
add openid login to backstage bubble

TODO: we probably want to hide this for the average user. This at
least makes it easy to access this login form.

cdent commented:
Does this just add it as an additional form alongside the existing one?

If so would it be possible to make it so you can toggle between the two?

jdlrobson commented:
That's what I meant by my TODO message. I guess toggling would be useful, but I guess some sort of stickiness would be useful - ie. if I always login via openid I'd always like that to be the open one.

cdent commented:
stickiness would require sessioning, which we don't want, and besides we've made it so cookies are hard to see from javascript

so i think a toggle, with the user/pass being the default (since that's what 99% of users use) would be right (and less visually noisy)

jdlrobson committed:
Merge branch 'master' of git://github.com/TiddlySpace/tiddlyspace into 838

jdlrobson committed:
only show one login form at a time

this gives both login forms class login-method
which is selected is chosen by which is the target element (if the hash
is #userpass-login the element with id userpass-login will be the
target and can be targetted via css .login-method#target like so

note if browser doesn't support :target at moment this means no login
form will show so fix needed...

jdlrobson committed:
old browser fallback

if :target is not supported then the active element will be
display none. We make sure this isn't the situation by showing any
elements in this situation.

jdlrobson commented:
I've tweaked this. Not checked the old browser fallback as don't have access to a dev version of TiddlySpace on Windows.. but don't believe it will be an issue.

cdent commented:
@jdlrobson it's still not working right, sorry

Now it has the same problem that the frontpage had when logging with openid: when you log in with openid it raises a bunch of alerts about trying to create a space.

Since you did the fix on the frontpage I assume you can suss out what's up here?
